STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm.c

```diff
@@ -61,27 +109,134 @@ AZ_NODISCARD az_result az_ulib_dm_install(az_span package_name)
   return result;
 }
 
-AZ_NODISCARD az_result az_ulib_dm_uninstall(az_span package_name)
+static az_result install_in_memory(void* base_address, az_span package_name)
 {
   _az_PRECONDITION_NOT_NULL(_az_dm_cb);
   az_result result;
 
-  if(az_span_is_content_equal(package_name, AZ_SPAN_FROM_STR("cipher_v1i1")))
+  if(get_package(package_name) != NULL)
   {
-    result = cipher_v1i1_destroy();
+    result = AZ_ERROR_ULIB_ELEMENT_DUPLICATE;
   }
-  else if(az_span_is_content_equal(package_name, AZ_SPAN_FROM_STR("cipher_v2i1")))
+  else
   {
-    result = cipher_v2i1_destroy();
+    uint32_t publish_position = *((uint32_t*)base_address + AZ_ULIB_DM_PACKAGE_PUBLISH);
+    publish_interface publish = (publish_interface)((uint8_t*)base_address + publish_position + 
+        (AZ_ULIB_DM_PACKAGE_PUBLISH << 2));
+    const az_ulib_ipc_vtable* vtable = az_ulib_ipc_get_vtable();
+    if((result = publish(vtable)) == AZ_OK)
+    {
+      _az_ulib_dm_package* package = get_first_package_free();
+      if(package == NULL)
+      {
+        result = AZ_ERROR_NOT_ENOUGH_SPACE;
+      }
+      else
+      {
+        az_span_copy(package->name, package_name);
+        package->name = az_span_create(az_span_ptr(package->name), az_span_size(package_name));
+        package->address = base_address;
+        result = AZ_OK;
+      }
+    }
   }
-  else if(az_span_is_content_equal(package_name, AZ_SPAN_FROM_STR("sprinkler_v1i1")))
+
+  return result;
+}
+
+static az_result install_from_blob(void* base_address, az_span package_name)
```

Comment:
the current implementation shares the IP instance from the main application connecting to hub. In this we will have to create our own which would be tricky. Can you include a void parameter to pass the memory location for the connection control block?

Reply:
The DM don't have the connection control block, we will need to fix it in a different way, let's chat about it.